testing/integration/datastreams_test.go

```diff
@@ -0,0 +1,361 @@
+//go:build integration
+
+package integration
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"net/http/httputil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"text/template"
+	"time"
+
+	"github.com/elastic/elastic-agent-libs/kibana"
+	"github.com/elastic/elastic-agent/pkg/control/v2/client"
+	atesting "github.com/elastic/elastic-agent/pkg/testing"
+	"github.com/elastic/elastic-agent/pkg/testing/define"
+	"github.com/elastic/elastic-agent/pkg/testing/tools"
+	"github.com/elastic/elastic-agent/pkg/testing/tools/estools"
+	"github.com/elastic/elastic-agent/version"
+	"github.com/stretchr/testify/require"
+)
+
+func TestFlattenedDatastreamFleetPolicy(t *testing.T) {
+	dsType := "logs"
+	dsNamespace := strings.ToLower(fmt.Sprintf("%snamespace%d", t.Name(), rand.Uint64()))
+	dsDataset := strings.ToLower(fmt.Sprintf("%s-dataset", t.Name()))
+	numEvents := uint64(60)
+
+	tempDir := t.TempDir()
+	logFilePath := filepath.Join(tempDir, "log.log")
+	generateLogFile(t, logFilePath, 2*time.Millisecond, numEvents)
+
+	info := define.Require(t, define.Requirements{
+		Local: false,
+		Stack: &define.Stack{
+			Version: version.Agent + "-SNAPSHOT",
```

Comment:
CI was failing because the stack was too old, so I "hardcoded" the version at least to make sure the tests are passing